python/frameworks/mcp/traceai_mcp/__init__.py

```
@@ -0,0 +1,203 @@
+            yield InstrumentedStreamReader(read_stream), InstrumentedStreamWriter(write_stream)
+
+    def _base_session_init_wrapper(
+        self, wrapped: Callable[..., None], instance: Any, args: Any, kwargs: Any
+    ) -> None:
+        wrapped(*args, **kwargs)
+        reader = getattr(instance, "_incoming_message_stream_reader", None)
+        writer = getattr(instance, "_incoming_message_stream_writer", None)
+        if reader and writer:
+            setattr(
+                instance, "_incoming_message_stream_reader", ContextAttachingStreamReader(reader)
+            )
```

Ruff (0.11.9)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)